lib/active_merchant/billing/gateways/securion_pay.rb

```diff
@@ -0,0 +1,269 @@
+
+      self.homepage_url = 'https://securionpay.com/'
+      self.display_name = 'SecurionPay'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        'incorrect_number' => STANDARD_ERROR_CODE[:incorrect_number],
+        'invalid_number' => STANDARD_ERROR_CODE[:invalid_number],
+        'invalid_expiry_month' => STANDARD_ERROR_CODE[:invalid_expiry_date],
+        'invalid_expiry_year' => STANDARD_ERROR_CODE[:invalid_expiry_date],
+        'invalid_cvc' => STANDARD_ERROR_CODE[:invalid_cvc],
+        'expired_card' => STANDARD_ERROR_CODE[:expired_card],
+        'insufficient_funds' => STANDARD_ERROR_CODE[:card_declined],
+        'incorrect_cvc' => STANDARD_ERROR_CODE[:incorrect_cvc],
+        'incorrect_zip' => STANDARD_ERROR_CODE[:incorrect_zip],
+        'card_declined' => STANDARD_ERROR_CODE[:card_declined],
+        'processing_error' => STANDARD_ERROR_CODE[:processing_error],
+        'lost_or_stolen' => STANDARD_ERROR_CODE[:card_declined],
+        'suspected_fraud' => STANDARD_ERROR_CODE[:card_declined],
+        'expired_token' => STANDARD_ERROR_CODE[:card_declined]
+      }
+
+      def initialize(options={})
+        requires!(options, :secret_key)
+        @secret_key = options[:secret_key]
+
+        super
+      end
+
+      def purchase(money, payment, options={})
+        post = {}
+        post = create_post_for_auth_or_purchase(money, payment, options)
+        commit(:post, 'charges', post, options)
+      end
+
+      def authorize(money, payment, options={})
+        post = {}
+        post = create_post_for_auth_or_purchase(money, payment, options)
+        post[:captured] = "false"
+        commit(:post, 'charges', post, options)
+      end
+
+      def capture(money, authorization, options = {})
+        post = {}
+        add_amount(post, money, options)
+
+        commit(:post, "charges/#{CGI.escape(authorization)}/capture", post, options)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        add_amount(post, money, options)
+        commit(:post, "charges/#{CGI.escape(authorization)}/refund", post, options)
+      end
+
+      def void(authorization, options = {})
+        commit(:post, "charges/#{CGI.escape(authorization)}/refund", {}, options)
+      end
+
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def store(payment, options = {})
+        post = {}
+        add_creditcard(post, payment, options)
+
+        post[:description] = options[:description] if options[:description]
+        post[:email] = options[:email] if options[:email]
+        if options[:customer]
+          update_customer(options[:customer], post, options)
+        else
+          MultiResponse.run(:first) do |r|
+            r.process { verify(payment, options) }
```

Comment:
Why is `store` calling `verify`? Can you ditch the MultiResponse and just make a single call to store a card?